shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/data/network/core/NetworkClientTest.kt

```diff
@@ -1,26 +1,57 @@
 package co.nimblehq.blisskmmic.data.network.core
 
-import co.nimblehq.blisskmmic.helpers.mock.NETWORK_MOCK_MODEL_RESULT
+import co.nimblehq.blisskmmic.BuildKonfig
+import co.nimblehq.blisskmmic.data.network.helpers.API_VERSION
 import co.nimblehq.blisskmmic.helpers.mock.NetworkMockModel
+import co.nimblehq.blisskmmic.helpers.mock.NETWORK_MOCK_MODEL_RESULT
 import co.nimblehq.blisskmmic.helpers.mock.ktor.jsonMockEngine
+import co.nimblehq.jsonapi.model.JsonApiException
+import io.kotest.matchers.collections.shouldContain
 import io.kotest.matchers.shouldBe
 import io.ktor.client.request.*
 import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
 import kotlin.test.Test
+import kotlin.test.fail
 
 @ExperimentalCoroutinesApi
 class NetworkClientTest {
 
-    val engine = jsonMockEngine(NETWORK_MOCK_MODEL_RESULT)
+    private val path = "user"
+    private val request = HttpRequestBuilder()
+
+    @BeforeTest
+    fun setUp() {
+        request.url("$BuildKonfig.BASE_URL$API_VERSION$path")
+    }
 
     @Test
-    fun testMockModel() = runTest {
+    fun `when calling fetch, it returns correct object`() = runTest {
+        val engine = jsonMockEngine(NETWORK_MOCK_MODEL_RESULT, path)
         val networkClient = NetworkClient(engine = engine)
         networkClient
-            .fetch<NetworkMockModel>(HttpRequestBuilder())
+            .fetch<NetworkMockModel>(request)
             .collect {
                 it.title shouldBe "Hello"
             }
     }
+
+    @Test
+    fun `when calling fetch with invalid path, it returns correct object`() = runTest {
+        val engine = jsonMockEngine(NETWORK_MOCK_MODEL_RESULT, "")
+        val networkClient = NetworkClient(engine = engine)
+        networkClient
+            .fetch<NetworkMockModel>(request)
+            .catch { error ->
```

Comment:
You might need to use this https://github.com/cashapp/turbine, it's recommended by Android doc to test flow.